src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -0,0 +1,139 @@
+
+    @Autowired
+    private AccessTokenStore accessTokenStore;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public void configure() {
+
+        /*
+          Rest endpoint to initiate payment for buy goods
+
+          Sample request body: {
+              "BusinessShortCode": 174379,
+              "Amount": 1,
+              "PartyA": 254708374149,
+              "PartyB": 174379,
+              "PhoneNumber": 254708374149,
+              "CallBackURL": "https://mydomain.com/path",
+              "AccountReference": "CompanyXLTD",
+              "TransactionDesc": "Payment of X"
+            }
+         */
+        from("rest:POST:/buygoods")
+                .id("buy-goods-online")
+                .process(exchange -> {
+                    String body = exchange.getIn().getBody(String.class);
+                    BuyGoodsPaymentRequestDTO buyGoodsPaymentRequestDTO = objectMapper.readValue(
+                            body, BuyGoodsPaymentRequestDTO.class);
+
+                    exchange.setProperty(BUY_GOODS_REQUEST_BODY, buyGoodsPaymentRequestDTO);
+                    logger.info(body);
+                    logger.info(buyGoodsPaymentRequestDTO.toString());
+
+                })
+                .to("direct:buy-goods-base");
+
+        /*
+         * Starts the payment flow
+         *
+         * Step1: Authenticate the user by initiating [get-access-token] flow
+         * Step2: On successful [Step1], directs to [lipana-buy-goods] flow
+         */
+        from("direct:buy-goods-base")
+                .id("buy-goods-base")
+                .log(LoggingLevel.INFO, "Starting buy goods flow")
+                .to("direct:get-access-token")
+                .process(exchange -> {
+                    accessTokenStore.setExpiresOn(3599);
+                    exchange.setProperty(ACCESS_TOKEN, accessTokenStore.getAccessToken());
+
+                })
+                .log(LoggingLevel.INFO, "Got access token, moving on to API call.")
+                .to("direct:lipana-buy-goods");
+
+        /*
+         * Takes the access toke and payment request and forwards the requests to lipana API.
+         * [Password] and [TransactionType] are set in runtime and request is forwarded to lipana endpoint.
+         */
+        from("direct:lipana-buy-goods")
+                .removeHeader("*")
+                .setHeader(Exchange.HTTP_METHOD, constant("POST"))
+                .setHeader("Content-Type", constant("application/json"))
+                .setHeader("Authorization", simple("Bearer ${exchangeProperty."+ACCESS_TOKEN+"}"))
+                .setBody(exchange -> {
+                    BuyGoodsPaymentRequestDTO buyGoodsPaymentRequestDTO =
+                            (BuyGoodsPaymentRequestDTO) exchange.getProperty(BUY_GOODS_REQUEST_BODY);
+
+
+                    String password = getPassword("" + buyGoodsPaymentRequestDTO.getBusinessShortCode(),
```

Comment:
Can you cross check the function arguments? Do we need to pass the passKey here? Your getPassword function is within the same class. It will have access to the passKey directly in src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java:L25

Reply:
Yes I got your point, I did that for the separation fo concern. Because with current implementation function only depends on passed arguments and nothing else. So whoever calls the function is responsible for meeting its arg requirements. In case you want the other please tell me Ill modify.

Reply:
If you want to keep SOC, move the function to a Utils class.

Reply:
okay Ill do that.

Reply:
done with this